Switch to reusable workflows for codeception, php-style, poedit…

robertSt7 opened the pull request.

robertSt7 committed:
Task: Switch to reusable workflows for codeception, php-style, poeditor-export and static-analysis.

robertSt7 committed:
Task: Switch to reusable workflows for codeception, php-style, poeditor-export and static-analysis.

blankse commented on .github/workflows/php-cs-fixer.yaml:
```diff
@@ -17,19 +17,9 @@ permissions:
-              with:
-                  commit_message: Apply php-cs-fixer changes
+    php-style:
+        uses: pimcore/workflows-collection-public/.github/workflows/reusable-php-style.yaml@main
```
@robertSt7 Do this workflow also support pull requests from forks? See https://github.com/pimcore/admin-ui-classic-bundle/pull/517

herbertroth replied:
@blankse It supports pull request from forks(now).

herbertroth committed:
Use workflows for public repos.

herbertroth committed:
Use workflows for public repos.

herbertroth committed:
Fix errors.

herbertroth committed:
Fix errors.

blankse commented on .github/workflows/php-cs-fixer.yaml:
```diff
@@ -17,19 +17,10 @@ permissions:
     contents: read
```
Must it be `write`?
```suggestion
    contents: write
```

blankse commented on .github/workflows/php-cs-fixer.yaml:
```diff
@@ -17,19 +17,10 @@ permissions:
+        secrets:
+            PHP_CS_FIXER_GITHUB_TOKEN: ${{ secrets.PHP_CS_FIXER_GITHUB_TOKEN }}
+        with:
+            head_ref: ${{ github.ref }}
+            config_file: .php-cs-fixer.dist.php
```
Is here `repository` missing?
```suggestion
            head_ref: ${{ github.event.pull_request.head.ref }}
            repository: ${{ github.event.pull_request.head.repo.full_name }}
            config_file: .php-cs-fixer.dist.php
```

blankse commented:
@robertSt7 @herbertroth My suggestions are from the file in `pimcore/pimcore`
https://github.com/pimcore/pimcore/blob/11.x/.github/workflows/php-cs-fixer.yaml

When I update my fork of `pimcore/pimcore` the php-cs-fixer action in the fork are failing:
https://github.com/blankse/pimcore/actions/runs/9748030425/job/26902090652

Is it possible to avoid this?

herbertroth committed:
Update CE to EE sync for 1.5 branch.

(cherry picked from commit 8a02c34d4f368752ac7fc2378d85e4f965b00f18)

herbertroth committed:
Use latest reusable workflows.
Static analysis and codeception test will run based on php versions in composer.json

Add sonar config and workflow file.

herbertroth committed:
Use latest reusable workflows.
Static analysis and codeception test will run based on php versions in composer.json

Add sonar config and workflow file.

herbertroth committed:
Merge remote-tracking branch 'origin/Switch-to-reusable-workflows-for-codeception,-php-style,-poeditor-export-and-static-analysis' into Switch-to-reusable-workflows-for-codeception,-php-style,-poeditor-export-and-static-analysis

herbertroth committed:
Use latest reusable workflows.
Static analysis and codeception test will run based on php versions in composer.json

Add sonar config and workflow file.

herbertroth committed:
Fix composer.json for tests.
